lib/AST/Decl.cpp

```diff
@@ -4132,7 +4132,8 @@ bool ClassDecl::hasMissingDesignatedInitializers() const {
   if (!Bits.ClassDecl.ComputedHasMissingDesignatedInitializers) {
     auto *mutableThis = const_cast<ClassDecl *>(this);
     mutableThis->Bits.ClassDecl.ComputedHasMissingDesignatedInitializers = 1;
-    (void)mutableThis->lookupDirect(DeclBaseName::createConstructor());
+    (void)mutableThis->lookupDirect(DeclBaseName::createConstructor(),
+                                    LookupDirectFlags::IgnoreNewExtensions);
```

Comment:
Looking this over, I'm not sure why it's passing in `IgnoreNewExtensions`. Maybe a comment about the "why"?

Reply:
Everywhere you see `IgnoreNewExtensions` satisfies the precondition in the docs for it.

> e.g. when it is known that a particular member declaration will never appear in an extension.

- Designated initializers are banned in extensions.
- `associatedtype` declarations outside of protocols are malformed
- `TypeCheckProtocol` is looking up requirements in the protocol just to pop a diagnostic and in both cases goes out of its way to filter out extensions.

The Clang Importer ones are the trickier ones to justify since that's not my domain at all.  But I think the idea is that the `init(rawValue:)` constructor will only ever be added to the class itself, and whatever an "anonymous inner field" is, it too will only appear in a struct's decl context.

Now that I've gone to the trouble of typing this out, I'll put in a follow-up patch that adds some nota benes.